symphony-bdk-core-invokers/symphony-bdk-core-invoker-api/src/main/java/com/symphony/bdk/core/api/invoker/ApiClientBuilder.java

```diff
@@ -0,0 +1,68 @@
+package com.symphony.bdk.core.api.invoker;
+
+/**
+ * Builder class to create a new instance of an {@link ApiClient}
+ */
+public interface ApiClientBuilder {
+  /**
+   * @return a new {@link ApiClient} based on the previously called methods below.
+   */
+  ApiClient buildClient();
+
+  /**
+   * @param basePath base URL to be used for HTTP calls
+   * @return the updated instance of {@link ApiClientBuilder}
+   */
+  ApiClientBuilder basePath(String basePath);
+
+  /**
+   * @param userAgent user agent value to be put in the User-Agent header
+   * @return the updated instance of {@link ApiClientBuilder}
+   */
+  ApiClientBuilder userAgent(String userAgent);
+
+  /**
+   * @param keyStoreBytes content of the key store (aka client certificate)
+   * @param keyStorePassword password of the key store
+   * @return the updated instance of {@link ApiClientBuilder}
+   */
+  ApiClientBuilder keyStore(byte[] keyStoreBytes, String keyStorePassword);
+
+  /**
+   * @param trustStoreBytes content of the trust store
+   * @param trustStorePassword password of the truststore
+   * @return the updated instance of {@link ApiClientBuilder}
+   */
+  ApiClientBuilder trustStore(byte[] trustStoreBytes, String trustStorePassword);
+
+  /**
+   * @param key name of the header to be added in all HTTP calls
+   * @param value header value
+   * @return the updated instance of {@link ApiClientBuilder}
+   */
+  ApiClientBuilder withDefaultHeader(String key, String value);
+
+  /**
+   * @param temporaryFolderPath temporary folder path where to download files sent as response
```

Comment:
Do we know what is the default value?

Reply:
Value is used here : https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-core-invokers/symphony-bdk-core-invoker-jersey2/src/main/java/com/symphony/bdk/core/api/invoker/jersey2/ApiClientJersey2.java#L528
If it is not set, it uses value of property "java.io.tmpdir" which depends on the OS : https://dzone.com/articles/working-with-temporary-filesfolders-in-java